### PR TITLE
LTS Maintainance

### PR DESCRIPTION
**Pull request by barbara79**
# Description

Closes #

# Checklist

- [x] Update release Notes
- [ ] Add to the current github milestone 
- [x] Update commercetools api reference

### CODE REVIEW

File: commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java [hunk 1]
```diff
@@ -154,6 +154,10 @@
  </ul>
  -->
  -->
+ <h3 class=released-version id="v2_8_0">2.8.0 (04.04.2022)</h3>
+ <ul>
+ <li class=new-in-release>Support new Order message: {@link io.sphere.sdk.orders.messages.OrderPaymentAddedMessage}</li>
```
Comment (jenschude) [inline commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java:159]: please add the other changes too

File: commercetools-models/src/main/java/io/sphere/sdk/apiclient/ApiClientDraftBuilder.java [hunk 2]
```diff
@@ -12,7 +12,11 @@
 public final class ApiClientDraftBuilder extends ApiClientDraftBuilderBase {
 
     ApiClientDraftBuilder(final @Nullable Integer deleteDaysAfterCreation, final String name, final String scope) {
-        super(deleteDaysAfterCreation, name, scope);
+        super(null, deleteDaysAfterCreation, name, null, scope);
+    }
+
+    ApiClientDraftBuilder(final @Nullable Long accessTokenValiditySecondsfinal, @Nullable Integer deleteDaysAfterCreation, final String name, final @Nullable Long refreshTokenValiditySeconds, final String scope) {
```
Comment (jenschude) [inline commercetools-models/src/main/java/io/sphere/sdk/apiclient/ApiClientDraftBuilder.java:18]: this constructor makes no sense as the additional parameters are not used

**Review by jenschude — CHANGES_REQUESTED** [review]

File: commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java (hunk 1 above)
Comment (barbara79) [reply commercetools-internal-docs/src/main/java/io/sphere/sdk/meta/ReleaseNotes.java:159]: you're right I just find it out ;) I will update with the next commit.
Thanks

File: commercetools-internal-docs/src/test/java/introspection/rules/ImplClassesAreForPackageScopeRule.java [hunk 3]
```diff
@@ -9,7 +9,7 @@
             asList(("ByIdDeleteCommandImpl,CommandImpl,CreateCommandImpl," +
                     "CustomObjectQueryImpl,ResourceImpl,ResourceQueryModelImpl," +
                     "ResourceViewImpl,GenericMessageImpl,MessageImpl," +
-                    "MetaModelGetDslImpl,MetaModelQueryDslImpl,MoneyImpl," +
+                    "MetaModelGetDslImpl,MetaModelHeadDslImplMetaModelQueryDslImpl,MoneyImpl," +
```
Comment (jenschude) [inline commercetools-internal-docs/src/test/java/introspection/rules/ImplClassesAreForPackageScopeRule.java:12]:
```suggestion
                    "MetaModelGetDslImpl,MetaModelQueryDslImpl,MetaModelHeadDslImplMetaModelQueryDslImpl,MoneyImpl," +
```

File: commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByIdHead.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import io.sphere.sdk.models.Identifiable;
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.products.expansion.ProductExpansionModel;
+import io.sphere.sdk.products.search.PriceSelectionRequestDsl;
+import io.sphere.sdk.queries.HeadDsl;
+import io.sphere.sdk.queries.MetaModelHeadDsl;
+
+import java.util.List;
+
+public interface ProductByIdHead extends MetaModelHeadDsl<Product, Product, ProductByIdHead, ProductExpansionModel<Product>>, PriceSelectionRequestDsl<ProductByIdHead> {
```
Comment (jenschude) [inline commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByIdHead.java:14]: The Head methods don't have expansion, price selection etc

File: commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByIdHeadImpl.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import io.sphere.sdk.queries.MetaModelHeadDslBuilder;
+import io.sphere.sdk.queries.MetaModelHeadDslImpl;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static io.sphere.sdk.products.search.PriceSelectionQueryParameters.extractPriceSelectionFromHttpQueryParameters;
+import static io.sphere.sdk.products.search.PriceSelectionQueryParameters.getQueryParametersWithPriceSelection;
+
+final class ProductByIdHeadImpl extends MetaModelHeadDslImpl<Product, Product, ProductByIdHead, ProductExpansionModel<Product>> implements ProductByIdHead {
```
Comment (jenschude) [inline commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByIdHeadImpl.java:16]: The Head methods don't have expansion, price selection etc

File: commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByKeyHead.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.products.expansion.ProductExpansionModel;
+import io.sphere.sdk.products.search.PriceSelectionRequestDsl;
+import io.sphere.sdk.queries.HeadDsl;
+import io.sphere.sdk.queries.MetaModelGetDsl;
+import io.sphere.sdk.queries.MetaModelHeadDsl;
+
+import java.util.List;
+
+public interface ProductByKeyHead extends MetaModelHeadDsl<Product, Product, ProductByKeyHead, ProductExpansionModel<Product>>, PriceSelectionRequestDsl<ProductByKeyHead> {
```
Comment (jenschude) [inline commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByKeyHead.java:14]: The Head methods don't have expansion, price selection etc

File: commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByKeyHeadImpl.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+import io.sphere.sdk.queries.MetaModelHeadDslImpl;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static io.sphere.sdk.client.SphereRequestUtils.urlEncode;
+import static io.sphere.sdk.products.search.PriceSelectionQueryParameters.extractPriceSelectionFromHttpQueryParameters;
+import static io.sphere.sdk.products.search.PriceSelectionQueryParameters.getQueryParametersWithPriceSelection;
+
+final class ProductByKeyHeadImpl extends MetaModelHeadDslImpl<Product, Product, ProductByKeyHead, ProductExpansionModel<Product>> implements ProductByKeyHead {
```
Comment (jenschude) [inline commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByKeyHeadImpl.java:19]: The Head methods don't have expansion, price selection etc

File: commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductsHead.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+import io.sphere.sdk.expansion.ExpansionPath;
+import io.sphere.sdk.products.Product;
+import io.sphere.sdk.products.expansion.ProductExpansionModel;
+import io.sphere.sdk.products.search.PriceSelectionRequestDsl;
+import io.sphere.sdk.queries.MetaModelHeadDsl;
+
+import java.util.List;
+
+public interface ProductsHead extends MetaModelHeadDsl<Product, Product, ProductsHead, ProductExpansionModel<Product>>, PriceSelectionRequestDsl<ProductsHead> {
```
Comment (jenschude) [inline commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductsHead.java:11]: The Head methods don't have expansion, price selection etc

File: commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductsHeadImpl.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+import io.sphere.sdk.queries.MetaModelHeadDslBuilder;
+import io.sphere.sdk.queries.MetaModelHeadDslImpl;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static io.sphere.sdk.products.search.PriceSelectionQueryParameters.extractPriceSelectionFromHttpQueryParameters;
+import static io.sphere.sdk.products.search.PriceSelectionQueryParameters.getQueryParametersWithPriceSelection;
+
+final class ProductsHeadImpl extends MetaModelHeadDslImpl<Product, Product, ProductsHead, ProductExpansionModel<Product>> implements ProductsHead {
```
Comment (jenschude) [inline commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductsHeadImpl.java:16]: The Head methods don't have expansion, price selection etc

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/queries/HeadDsl.java [hunk 10]
```diff
@@ -0,0 +1,7 @@
+package io.sphere.sdk.queries;
+
+import io.sphere.sdk.expansion.ReferenceExpansionDsl;
+
+public interface HeadDsl<R, T, C> extends Head<R>, ReferenceExpansionDsl<T, C> {
```
Comment (jenschude) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/queries/HeadDsl.java:5]: No reference expansion possible with head method

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/queries/MetaModelHeadDsl.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import io.sphere.sdk.expansion.MetaModelReferenceExpansionDsl;
+
+/**
+ *
+ * @param <R> type of the query result
+ * @param <T> unit of the query result
+ * @param <C> type of the class implementing this class
+ * @param <E> type of the expansion model
+ */
+public interface MetaModelHeadDsl<R, T, C extends MetaModelHeadDsl<R, T, C, E>, E> extends HeadDsl<R, T, C>, MetaModelReferenceExpansionDsl<T, C, E> {
```
Comment (jenschude) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/queries/MetaModelHeadDsl.java:12]: no reference expansion possible

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/queries/MetaModelHeadDslBuilder.java [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.function.Function;
+
+/**
+ *
+ * @param <R> result type, maybe directly {@code T} or sth. like {@code List<T>}
+ * @param <T> type a single result item
+ * @param <C> type of the class implementing this class
+ * @param <E> type of the expansion model
+ */
+public final class MetaModelHeadDslBuilder<R, T, C extends MetaModelHeadDsl<R, T, C, E>, E> extends Base implements Builder<C> {
```
Comment (jenschude) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/queries/MetaModelHeadDslBuilder.java:19]: no reference expansion with head possible

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/queries/MetaModelHeadDslImpl.java [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+import static io.sphere.sdk.utils.SphereInternalUtils.listOf;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * @param <R> result type, maybe directly {@code T} or sth. like {@code List<T>}
+ * @param <T> type of the query result
+ * @param <C> type of the class implementing this class
+ * @param <E> type of the expansion model
+ */
+public abstract class MetaModelHeadDslImpl<R, T, C extends MetaModelHeadDsl<R, T, C, E>, E> extends Base implements MetaModelHeadDsl<R, T, C, E>, MetaModelExpansionDslExpansionModelRead<T, C, E> {
```
Comment (jenschude) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/queries/MetaModelHeadDslImpl.java:35]: no expansion model necessary

**Review by jenschude — CHANGES_REQUESTED** [review]

File: commercetools-internal-docs/src/test/java/introspection/rules/ImplClassesAreForPackageScopeRule.java (hunk 3 above)
Comment (barbara79) [reply commercetools-internal-docs/src/test/java/introspection/rules/ImplClassesAreForPackageScopeRule.java:12]: oh I forgot the comma! thanks :D

Comment (jenschude) [reply commercetools-internal-docs/src/test/java/introspection/rules/ImplClassesAreForPackageScopeRule.java]: No you removed one MetaModel ;)

File: commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductsHeadImpl.java (hunk 9 above)
Comment (barbara79) [reply commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductsHeadImpl.java:16]: thanks I followed the wrong PR

File: commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByKeyHeadImpl.java (hunk 7 above)
Comment (barbara79) [reply commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByKeyHeadImpl.java:19]: thanks, I followed the wrong PR about the price selection

File: commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByIdHeadImpl.java (hunk 5 above)
Comment (barbara79) [reply commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByIdHeadImpl.java]: thanks, I followed the wrong PR about the price selection

File: commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByIdHead.java (hunk 4 above)
Comment (barbara79) [reply commercetools-models/src/main/java/io/sphere/sdk/products/queries/ProductByIdHead.java]: thanks, I followed the wrong PR about the price selection

File: commercetools-internal-docs/src/test/java/introspection/rules/ImplClassesAreForPackageScopeRule.java [hunk 14]
```diff
@@ -9,7 +9,7 @@
             asList(("ByIdDeleteCommandImpl,CommandImpl,CreateCommandImpl," +
                     "CustomObjectQueryImpl,ResourceImpl,ResourceQueryModelImpl," +
                     "ResourceViewImpl,GenericMessageImpl,MessageImpl," +
-                    "MetaModelGetDslImpl,MetaModelQueryDslImpl,MoneyImpl," +
+                    "MetaModelGetDslImpl,MetaModelQueryDslImpl,MetaModelHeadDslImpl,MetaModelQueryDslImpl,MoneyImpl," +
```
Comment (jenschude) [inline commercetools-internal-docs/src/test/java/introspection/rules/ImplClassesAreForPackageScopeRule.java:12]:
```suggestion
                    "MetaModelGetDslImpl,MetaModelQueryDslImpl,MetaModelHeadDslImpl,MoneyImpl," +
```

duplicate entry

**Review by jenschude — APPROVED** [review]
small change

**Pull request merged**